### ISSUE TITLE
Adding index on string field breaks CONTAINS operator

### ISSUE THREAD

**Issue by obi1kenobi** [orientechnologies/orientdb#6953]
## OrientDB Version, operating system, or hardware.
- v2.0 SNAPSHOT[ ] - .18[ ] .17[ ] .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.1 SNAPSHOT[ ] - .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.2 SNAPSHOT[ ] - .12[x] .rc1[ ] .beta2[ ] .beta1[ ]

## Operating System
- [x] Linux
- [x] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

## Expected behavior and actual behavior
Expected: adding an index on a field should not change the returned results.

Actual: adding an index on a string field makes the `CONTAINS` operator ignore fields that should match it.

## Steps to reproduce the problem
```
CREATE CLASS Foo EXTENDS V
CREATE PROPERTY Foo.name String

CREATE VERTEX Foo SET name = 'foo'

// Both of the following queries should (and do) return the above Foo vertex.
SELECT * FROM Foo WHERE name IN ['foo', 'bar']
SELECT * FROM Foo WHERE ['foo', 'bar'] CONTAINS name

// Let's create the index.
CREATE INDEX Foo.name UNIQUE_HASH_INDEX

// Now, only the first query returns the correct result.
// The second query returns no results.
SELECT * FROM Foo WHERE name IN ['foo', 'bar']
SELECT * FROM Foo WHERE ['foo', 'bar'] CONTAINS name
```

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

Thanks for reporting, adding to my TODO list

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

Fixed in branch 2.2.x, the fix will be available in v 2.2.14

Thanks

Luigi
